File: valid parentheses/valid parentheses.go
func isValid(s string) bool {
    stack := make([]rune, 0);
    m := map[rune]rune {
        ')': '(',
        ']': '[',
        '}': '{',
    }

    for _, c := range s {
        switch c {
            case '(', '{', '[':
                stack = append(stack, c);
            case ')', '}', ']':
                if len(stack) == 0 || m[c] != stack[len(stack) -1] {
                    return false;
                } 
                stack = stack[:len(stack)-1]
        }
    }

    return len(stack) == 0
}

File: leetcode/a-easy/valid parentheses/valid parentheses.ts
function isValid(s: string): boolean {
    const matching: {[key: string]: string} = {
        "(" : ")",
        "{" : "}",
        "[" : "]", 
    }

    let stack: string[] = [];
    for (const char of s) {
        console.log(char);
        if (matching[char]) stack.push(char);
        else if (char !== matching[stack[stack.length -1]]) return false;
        else stack.pop();
    }

    return stack.length === 0;
};
